src/util/platform.rs

```diff
@@ -0,0 +1,648 @@
+use std::env;
+use std::ffi::OsString;
+use std::path::{Path, PathBuf};
+
+use color_eyre::eyre::bail;
+use color_eyre::eyre::WrapErr;
+use color_eyre::Result;
+use tracing::{debug, info, warn};
+
+use crate::commands;
+use crate::installable::Installable;
+
+/// Resolves an Installable from an environment variable, or falls back to the provided one.
+pub fn resolve_env_installable(var: &str, fallback: Installable) -> Installable {
```

Comment:
Bikeshed: this should return an `Option<Installable>`, as the callsites favour doing `platform::resolve_env_installable("NH_OS_FLAKE").unwrap_or(self.installable)`